modules/artifact-activation/src/main/java/org/mule/runtime/module/artifact/activation/api/ast/ArtifactAstUtils.java

```diff
@@ -107,43 +147,106 @@ public static ArtifactAst parseAndBuildAppExtensionModel(String[] configResource
   public static Optional<ExtensionModel> parseArtifactExtensionModel(ArtifactAst ast,
                                                                      ClassLoader artifactClassLoader,
                                                                      MuleContext muleContext) {
-
-    if (ast.topLevelComponentsStream()
-        .noneMatch(component -> APPLICATION_COMPONENT_TYPES.contains(component.getComponentType()))) {
+    if (!definesOwnExtensionModel(ast)) {
       return empty();
     }
 
-    Optional<ArtifactCoordinates> artifactCoordinates = muleContext.getConfiguration().getArtifactCoordinates();
+    return parseArtifactExtensionModel(ast,
+                                       artifactClassLoader,
+                                       muleContext.getConfiguration().getId(),
+                                       muleContext.getConfiguration().getArtifactCoordinates()
+                                           .map(ArtifactCoordinates::getVersion),
+                                       muleContext.getExtensionManager().getExtensions());
+  }
 
-    if (!artifactCoordinates.isPresent()) {
-      logModelNotGenerated("No version specified on muleContext", muleContext);
+  /**
+   * If the {@code ast} represents an artifact which defines reusable components (operations, sources, etc), it returns an
+   * {@link ExtensionModel} which represents it.
+   *
+   * @param ast                 the artifact's AST
+   * @param artifactClassLoader the artifact's classloader
+   * @param artifactId          the artifact's ID.
+   * @param artifactVersion     the artifact's version.
+   * @param dependencies        the dependencies in context.
+   * @return an optional {@link ExtensionModel}
+   */
+  public static Optional<ExtensionModel> parseArtifactExtensionModel(ArtifactAst ast,
+                                                                     ClassLoader artifactClassLoader,
+                                                                     String artifactId,
+                                                                     Optional<String> artifactVersion,
+                                                                     Set<ExtensionModel> dependencies) {
+    if (!artifactVersion.isPresent()) {
+      logModelNotGenerated("No version specified", artifactId, ast.getArtifactType());
```

Comment:
shouldn't it fail rather than returning empty?

Reply:
Similarly, I didn't want to change the [previous behavior](https://github.com/mulesoft/mule/blob/f3ad65b953a6af055a0aeee06e7c9bd19929c5b9/modules/artifact-activation/src/main/java/org/mule/runtime/module/artifact/activation/api/ast/ArtifactAstUtils.java#L120) to avoid breaking other unrelated things with this PR.
We could review that separately.